tutorials/arvo/cli-tutorial.md

```diff
@@ -0,0 +1,463 @@
+`(shoe:shoe command)` app core,
+pass that into a gate that transforms it into a `(agent:shoe command)` core, which is then cast
+as an `agent:gall` and passed into the `agent:dbug` gate, which
+endows the resulting core (stil a valid Gall agent core) with additional arms
+useful for debugging purposes. Finally, the `verb` gate (imported with `/+
+verb`) is called, which allows the agent to print what it is doing.
+
+```hoon
+|_  =bowl:gall
++*  this  .
+    def   ~(. (default-agent this %|) bowl)
+    des   ~(. (default:shoe this command) bowl)
+::
+```
+This is boilerplate Gall agent core code. We set `def` to be an alias for the part of the
+context where the `default-agent` lives, and set `des`  to be an alias for the part of the
+context where `shoe` library commands live.
+
+Next we begin implementing all of the arms 
+
+```hoon
+++  on-init   on-init:def
+++  on-save   !>(state)
+++  on-load
+  |=  old=vase
+  ^-  (quip card _this)
+  [~ this]
+::
+++  on-poke   on-poke:def
+++  on-watch  on-watch:def
+++  on-leave  on-leave:def
+++  on-peek   on-peek:def
+++  on-agent  on-agent:def
+++  on-arvo   on-arvo:def
+++  on-fail   on-fail:def
+```
+Boilerplate Gall app arms using the minimum implementation found in `def`.
+
+Here begins the implementations of the additional arms required by the
+`(shoe:shoe command)` interface.
+```hoon
+++  command-parser
+  |=  sole-id=@ta
+  ^+  |~(nail *(like [? command]))
+  (cold [& ~] (jest 'demo'))
+```
+`+command-parser` is of central importance - it is what is used to parse user
+input and transform it into `$command`s for the app to execute. Writing a proper
+command parser requires understanding of the Hoon parsing functions found in the
+standard library, but we will cover the bare essentials of parsing here.
+
+`+command-parser` is a gate which takes in a `sole-id=@ta` identifying a `sole`
+session and produces a particular sort of gate which takes in a `nail` (parse
+input) and returns an `edge` (parse output) that we call a parser or `$rule`.
+`+command-parser` produces a specific type of `$rule` whose result is of the
+type `[? command]`. Let us elaborate.
+
+ `nail` is the remainder of a parse given by `[p=hair q=tape]` where the
+`hair=[p=@ud q=@ud]` represent line and column of the current position of the parse and the
+`tape` is the parsing input. An `edge` is a given by `[p=hair q=(unit
+[p=* q=nail])]`. `p` says what we have parsed up to this point. `q` is the
+result of the parse, which is `~` if nothing valid has yet been parsed or a noun
+`p.q` that here will be a value of the `$command` type and `q.q` a `nail` that
+contains results that may still be unparsed. Here the
+`edge` is cast with `*(like [? command])`. `like [? command]`
+produces a generic `edge` which returns a result that is a cell of values of the
+`?` and `command` types - this is what forces `+command-parser` to produce a
+`$rule` that returns a value of type `[? command]`.
```

Comment:
These are not "the bare essential" lol. I think it's fine to handwave this away as "this is the type for a parser that produces a `[? command]`-shaped noun, see this parsing doc (TODO) for deets".

Reply:
I agree on wiping the parsing section. I should have a PR up for parsing docs next week anyways that I'll be able to link to from here.

https://github.com/urbit/docs/pull/912/commits/cae94e6de188b62721bb40412c37bf52b0aca6fc#diff-6d8f36f0b87617684df85577b6a7144dR339